fdbclient/ActorLineageProfiler.cpp

```diff
@@ -201,3 +194,86 @@ std::shared_ptr<Sample> SampleCollectorT::collect() {
 	packer.pack(res);
 	return packer.done(time);
 }
+
+void SampleCollection_t::refresh() {
+	auto sample = _collector->collect();
+	auto min = sample->time - windowSize;
+	double oldest = 0.0;
+	while (oldest < min && !data.empty()) {
+		// we remove at most 10 elements at a time. This is so we don't block the main thread for too long.
+		{
+			Lock _{ mutex };
+			int i = 0;
+			do {
+				oldest = data.front()->time;
+				data.pop_front();
```

Comment:
I think these lines should probably be reversed, otherwise you could end up deleting one too many samples (and you will need a conditional to check `data.size()` :/)